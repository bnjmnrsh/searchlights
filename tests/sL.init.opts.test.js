'use strict'
//github.com/suchipi/test-it
import searchLights from '../src/searchlights.js'

// console.log(searchLights._init().settings)

describe('_init() with options, no aSrchLtElsOpts', () => {
    // document.body.innerHTML = `<div></div>`
    it('the sL.settings object should  use _Default.aSrchLtElsOpts to populate els options', () => {
        const opts = {
            options: {
                blur: 2,
                dia: 200,
                blend: 'difference',
                opacity: 0.2,
                easing: 'ease',
                timing: 20,
                zindex: 1,
            },
        }

        expect(searchLights._init(opts)).toMatchSnapshot()
    })
})
