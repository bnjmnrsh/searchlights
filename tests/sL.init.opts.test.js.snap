// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_init() default output should accept an array of element options 1`] = `
Object {
  "_aDOMhadEls": Array [],
  "_build": [Function],
  "_destroy": [Function],
  "_init": [Function],
  "_nSrchLtsParent": <body
    class="mix-blend-mode"
  >
    <canvas
      class="blue srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,0,255)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="475"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease-out; transition-duration: 475ms;"
      width="106"
    />
    <canvas
      class="green srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,255,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="425"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease-out; transition-duration: 425ms;"
      width="106"
    />
    <canvas
      class="red srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(255,0,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="400"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease-out; transition-duration: 400ms;"
      width="106"
    />
    <div />
  </body>,
  "_nlSrchLtsEls": NodeList [
    <canvas
      class="blue srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,0,255)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="475"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease-out; transition-duration: 475ms;"
      width="106"
    />,
    <canvas
      class="green srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,255,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="425"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease-out; transition-duration: 425ms;"
      width="106"
    />,
    <canvas
      class="red srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(255,0,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="400"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease-out; transition-duration: 400ms;"
      width="106"
    />,
  ],
  "bEnableShowHide": true,
  "bUseInlineStyles": true,
  "fnPtrEnterCallbk": [Function],
  "fnPtrLeaveCallbk": [Function],
  "fnPtrMoveCallbk": [Function],
  "m": Object {
    "_fnDbnce": [Function],
    "fnCenterOnPtr": [Function],
    "fnCreateCtx": [Function],
    "fnCreateSrchLtEls": [Function],
    "fnDraw": [Function],
    "fnEventsCreate": [Function],
    "fnEventsDestroy": [Function],
    "fnFollowPtr": [Function],
    "fnSrchLtElsHide": [Function],
    "fnSrchLtElsShow": [Function],
    "fnStringifyClassArray": [Function],
  },
  "sParentEl": "body",
  "sTargetClass": ".searchlight",
  "settings": Object {
    "aSrchLtElsOpts": Array [
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "red",
          "srchLts-def",
          "searchlight",
          "searchlight",
          "searchlight",
          "searchlight",
        ],
        "color": "rgb(255,0,0)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 400,
      },
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "green",
          "srchLts-def",
          "searchlight",
          "searchlight",
          "searchlight",
          "searchlight",
        ],
        "color": "rgb(0,255,0)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 425,
      },
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "blue",
          "srchLts-def",
          "searchlight",
          "searchlight",
          "searchlight",
          "searchlight",
        ],
        "color": "rgb(0,0,255)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 475,
      },
    ],
    "blend": "screen",
    "blur": 3,
    "color": undefined,
    "dia": 100,
    "easing": "ease-out",
    "height": undefined,
    "opacity": 0.8,
    "timing": 90,
    "width": undefined,
    "zindex": 0,
  },
}
`;

exports[`_init() default output should return basic values 1`] = `
Object {
  "_aDOMhadEls": Array [],
  "_build": [Function],
  "_destroy": [Function],
  "_init": [Function],
  "_nSrchLtsParent": <body
    class="mix-blend-mode"
  >
    <canvas
      class="blue srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,0,255)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="475"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease-out; transition-duration: 475ms;"
      width="106"
    />
    <canvas
      class="green srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,255,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="425"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease-out; transition-duration: 425ms;"
      width="106"
    />
    <canvas
      class="red srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(255,0,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="400"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease-out; transition-duration: 400ms;"
      width="106"
    />
    <div />
  </body>,
  "_nlSrchLtsEls": NodeList [
    <canvas
      class="blue srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,0,255)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="475"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease-out; transition-duration: 475ms;"
      width="106"
    />,
    <canvas
      class="green srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,255,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="425"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease-out; transition-duration: 425ms;"
      width="106"
    />,
    <canvas
      class="red srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(255,0,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="400"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease-out; transition-duration: 400ms;"
      width="106"
    />,
  ],
  "bEnableShowHide": true,
  "bUseInlineStyles": true,
  "fnPtrEnterCallbk": [Function],
  "fnPtrLeaveCallbk": [Function],
  "fnPtrMoveCallbk": [Function],
  "m": Object {
    "_fnDbnce": [Function],
    "fnCenterOnPtr": [Function],
    "fnCreateCtx": [Function],
    "fnCreateSrchLtEls": [Function],
    "fnDraw": [Function],
    "fnEventsCreate": [Function],
    "fnEventsDestroy": [Function],
    "fnFollowPtr": [Function],
    "fnSrchLtElsHide": [Function],
    "fnSrchLtElsShow": [Function],
    "fnStringifyClassArray": [Function],
  },
  "sParentEl": "body",
  "sTargetClass": ".searchlight",
  "settings": Object {
    "aSrchLtElsOpts": Array [
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "red",
          "srchLts-def",
          "searchlight",
          "searchlight",
          "searchlight",
        ],
        "color": "rgb(255,0,0)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 400,
      },
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "green",
          "srchLts-def",
          "searchlight",
          "searchlight",
          "searchlight",
        ],
        "color": "rgb(0,255,0)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 425,
      },
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "blue",
          "srchLts-def",
          "searchlight",
          "searchlight",
          "searchlight",
        ],
        "color": "rgb(0,0,255)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 475,
      },
    ],
    "blend": "screen",
    "blur": 3,
    "color": undefined,
    "dia": 100,
    "easing": "ease-out",
    "height": undefined,
    "opacity": 0.8,
    "timing": 90,
    "width": undefined,
    "zindex": 0,
  },
}
`;

exports[`_init() default output should should be settable with options 1`] = `
Object {
  "_aDOMhadEls": Array [],
  "_build": [Function],
  "_destroy": [Function],
  "_init": [Function],
  "_nSrchLtsParent": <body
    class="mix-blend-mode"
  >
    <canvas
      class="blue srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,0,255)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="475"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 475ms;"
      width="106"
    />
    <canvas
      class="green srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,255,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="425"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 425ms;"
      width="106"
    />
    <canvas
      class="red srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(255,0,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="400"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 400ms;"
      width="106"
    />
    <div />
  </body>,
  "_nlSrchLtsEls": NodeList [
    <canvas
      class="blue srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,0,255)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="475"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 475ms;"
      width="106"
    />,
    <canvas
      class="green srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,255,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="425"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 425ms;"
      width="106"
    />,
    <canvas
      class="red srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(255,0,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="400"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 400ms;"
      width="106"
    />,
  ],
  "bEnableShowHide": true,
  "bUseInlineStyles": true,
  "fnPtrEnterCallbk": [Function],
  "fnPtrLeaveCallbk": [Function],
  "fnPtrMoveCallbk": [Function],
  "m": Object {
    "_fnDbnce": [Function],
    "fnCenterOnPtr": [Function],
    "fnCreateCtx": [Function],
    "fnCreateSrchLtEls": [Function],
    "fnDraw": [Function],
    "fnEventsCreate": [Function],
    "fnEventsDestroy": [Function],
    "fnFollowPtr": [Function],
    "fnSrchLtElsHide": [Function],
    "fnSrchLtElsShow": [Function],
    "fnStringifyClassArray": [Function],
  },
  "options": Object {
    "blend": "difference",
    "blur": 2,
    "dia": 200,
    "easing": "ease",
    "opacity": 0.2,
    "timing": 20,
    "zindex": 1,
  },
  "sParentEl": "body",
  "sTargetClass": ".searchlight",
  "settings": Object {
    "aSrchLtElsOpts": Array [
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "red",
          "srchLts-def",
          "searchlight",
        ],
        "color": "rgb(255,0,0)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 400,
      },
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "green",
          "srchLts-def",
          "searchlight",
        ],
        "color": "rgb(0,255,0)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 425,
      },
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "blue",
          "srchLts-def",
          "searchlight",
        ],
        "color": "rgb(0,0,255)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 475,
      },
    ],
    "blend": "difference",
    "blur": 2,
    "color": undefined,
    "dia": 200,
    "easing": "ease",
    "height": undefined,
    "opacity": 0.2,
    "timing": 20,
    "width": undefined,
    "zindex": 1,
  },
}
`;

exports[`_init() output with options object only, no els, should use _Default.aSrchLtElsOpts els should should be settable with options 1`] = `
Object {
  "_aDOMhadEls": Array [],
  "_build": [Function],
  "_destroy": [Function],
  "_init": [Function],
  "_nSrchLtsParent": <body
    class="mix-blend-mode"
  >
    <canvas
      class="blue srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,0,255)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="475"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 475ms;"
      width="106"
    />
    <canvas
      class="green srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,255,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="425"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 425ms;"
      width="106"
    />
    <canvas
      class="red srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(255,0,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="400"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 400ms;"
      width="106"
    />
    <div />
  </body>,
  "_nlSrchLtsEls": NodeList [
    <canvas
      class="blue srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,0,255)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="475"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 475ms;"
      width="106"
    />,
    <canvas
      class="green srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,255,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="425"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 425ms;"
      width="106"
    />,
    <canvas
      class="red srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(255,0,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="400"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 400ms;"
      width="106"
    />,
  ],
  "bEnableShowHide": true,
  "bUseInlineStyles": true,
  "fnPtrEnterCallbk": [Function],
  "fnPtrLeaveCallbk": [Function],
  "fnPtrMoveCallbk": [Function],
  "m": Object {
    "_fnDbnce": [Function],
    "fnCenterOnPtr": [Function],
    "fnCreateCtx": [Function],
    "fnCreateSrchLtEls": [Function],
    "fnDraw": [Function],
    "fnEventsCreate": [Function],
    "fnEventsDestroy": [Function],
    "fnFollowPtr": [Function],
    "fnSrchLtElsHide": [Function],
    "fnSrchLtElsShow": [Function],
    "fnStringifyClassArray": [Function],
  },
  "options": Object {
    "blend": "difference",
    "blur": 2,
    "dia": 200,
    "easing": "ease",
    "opacity": 0.2,
    "timing": 20,
    "zindex": 1,
  },
  "sParentEl": "body",
  "sTargetClass": ".searchlight",
  "settings": Object {
    "aSrchLtElsOpts": Array [
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "red",
          "srchLts-def",
          "searchlight",
        ],
        "color": "rgb(255,0,0)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 400,
      },
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "green",
          "srchLts-def",
          "searchlight",
        ],
        "color": "rgb(0,255,0)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 425,
      },
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "blue",
          "srchLts-def",
          "searchlight",
        ],
        "color": "rgb(0,0,255)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 475,
      },
    ],
    "blend": "difference",
    "blur": 2,
    "color": undefined,
    "dia": 200,
    "easing": "ease",
    "height": undefined,
    "opacity": 0.2,
    "timing": 20,
    "width": undefined,
    "zindex": 1,
  },
}
`;

exports[`_init() sL.settings with options object only, no els, should use _Default.aSrchLtElsOpts els should should be settable with options 1`] = `
Object {
  "_aDOMhadEls": Array [],
  "_build": [Function],
  "_destroy": [Function],
  "_init": [Function],
  "_nSrchLtsParent": <body
    class="mix-blend-mode"
  >
    <canvas
      class="blue srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,0,255)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="475"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 475ms;"
      width="106"
    />
    <canvas
      class="green srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,255,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="425"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 425ms;"
      width="106"
    />
    <canvas
      class="red srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(255,0,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="400"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 400ms;"
      width="106"
    />
    <div />
  </body>,
  "_nlSrchLtsEls": NodeList [
    <canvas
      class="blue srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,0,255)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="475"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 475ms;"
      width="106"
    />,
    <canvas
      class="green srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(0,255,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="425"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 425ms;"
      width="106"
    />,
    <canvas
      class="red srchLts-def searchlight"
      data-blend="screen"
      data-blur="3"
      data-color="rgb(255,0,0)"
      data-dia="100"
      data-opacity="0.8"
      data-timing="400"
      height="106"
      style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 400ms;"
      width="106"
    />,
  ],
  "bEnableShowHide": true,
  "bUseInlineStyles": true,
  "fnPtrEnterCallbk": [Function],
  "fnPtrLeaveCallbk": [Function],
  "fnPtrMoveCallbk": [Function],
  "m": Object {
    "_fnDbnce": [Function],
    "fnCenterOnPtr": [Function],
    "fnCreateCtx": [Function],
    "fnCreateSrchLtEls": [Function],
    "fnDraw": [Function],
    "fnEventsCreate": [Function],
    "fnEventsDestroy": [Function],
    "fnFollowPtr": [Function],
    "fnSrchLtElsHide": [Function],
    "fnSrchLtElsShow": [Function],
    "fnStringifyClassArray": [Function],
  },
  "options": Object {
    "blend": "difference",
    "blur": 2,
    "dia": 200,
    "easing": "ease",
    "opacity": 0.2,
    "timing": 20,
    "zindex": 1,
  },
  "sParentEl": "body",
  "sTargetClass": ".searchlight",
  "settings": Object {
    "aSrchLtElsOpts": Array [
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "red",
          "srchLts-def",
          "searchlight",
        ],
        "color": "rgb(255,0,0)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 400,
      },
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "green",
          "srchLts-def",
          "searchlight",
        ],
        "color": "rgb(0,255,0)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 425,
      },
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "blue",
          "srchLts-def",
          "searchlight",
        ],
        "color": "rgb(0,0,255)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 475,
      },
    ],
    "blend": "difference",
    "blur": 2,
    "color": undefined,
    "dia": 200,
    "easing": "ease",
    "height": undefined,
    "opacity": 0.2,
    "timing": 20,
    "width": undefined,
    "zindex": 1,
  },
}
`;

exports[`_init() with options, no aSrchLtElsOpts the sL.settings object should  use _Default.aSrchLtElsOpts to populate els options 1`] = `
Object {
  "_build": [Function],
  "_data": Object {
    "_aDOMhadEls": Array [],
    "_nSrchLtsParent": <body
      class="mix-blend-mode"
    >
      <canvas
        class="blue srchLts-def searchlight"
        data-blend="screen"
        data-blur="3"
        data-color="rgb(0,0,255)"
        data-dia="100"
        data-opacity="0.8"
        data-timing="475"
        height="106"
        hidden=""
        style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 475ms; z-index: 1;"
        width="106"
      />
      <canvas
        class="green srchLts-def searchlight"
        data-blend="screen"
        data-blur="3"
        data-color="rgb(0,255,0)"
        data-dia="100"
        data-opacity="0.8"
        data-timing="425"
        height="106"
        hidden=""
        style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 425ms; z-index: 1;"
        width="106"
      />
      <canvas
        class="red srchLts-def searchlight"
        data-blend="screen"
        data-blur="3"
        data-color="rgb(255,0,0)"
        data-dia="100"
        data-opacity="0.8"
        data-timing="400"
        height="106"
        hidden=""
        style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 400ms; z-index: 1;"
        width="106"
      />
    </body>,
    "_nlSrchLtsEls": NodeList [
      <canvas
        class="blue srchLts-def searchlight"
        data-blend="screen"
        data-blur="3"
        data-color="rgb(0,0,255)"
        data-dia="100"
        data-opacity="0.8"
        data-timing="475"
        height="106"
        hidden=""
        style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 475ms; z-index: 1;"
        width="106"
      />,
      <canvas
        class="green srchLts-def searchlight"
        data-blend="screen"
        data-blur="3"
        data-color="rgb(0,255,0)"
        data-dia="100"
        data-opacity="0.8"
        data-timing="425"
        height="106"
        hidden=""
        style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 425ms; z-index: 1;"
        width="106"
      />,
      <canvas
        class="red srchLts-def searchlight"
        data-blend="screen"
        data-blur="3"
        data-color="rgb(255,0,0)"
        data-dia="100"
        data-opacity="0.8"
        data-timing="400"
        height="106"
        hidden=""
        style="transform: translate3d(-53px, -53px, 0px); mix-blend-mode: screen; transition-timing-function: ease; transition-duration: 400ms; z-index: 1;"
        width="106"
      />,
    ],
  },
  "_destroy": [Function],
  "_init": [Function],
  "bEnableShowHide": true,
  "bUseInlineStyles": true,
  "fnPtrEnterCallbk": [Function],
  "fnPtrLeaveCallbk": [Function],
  "fnPtrMoveCallbk": [Function],
  "m": Object {
    "_fnDbnce": [Function],
    "fnCenterOnPtr": [Function],
    "fnCreateCtx": [Function],
    "fnCreateSrchLtEls": [Function],
    "fnDraw": [Function],
    "fnEventsCreate": [Function],
    "fnEventsDestroy": [Function],
    "fnFollowPtr": [Function],
    "fnSrchLtElsHide": [Function],
    "fnSrchLtElsShow": [Function],
    "fnStringifyClassArray": [Function],
  },
  "options": Object {
    "blend": "difference",
    "blur": 2,
    "dia": 200,
    "easing": "ease",
    "opacity": 0.2,
    "timing": 20,
    "zindex": 1,
  },
  "sParentEl": "body",
  "sTargetClass": ".searchlight",
  "settings": Object {
    "aSrchLtElsOpts": Array [
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "red",
          "srchLts-def",
        ],
        "color": "rgb(255,0,0)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 400,
      },
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "green",
          "srchLts-def",
        ],
        "color": "rgb(0,255,0)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 425,
      },
      Object {
        "blend": "screen",
        "blur": 3,
        "classes": Array [
          "blue",
          "srchLts-def",
        ],
        "color": "rgb(0,0,255)",
        "dia": 100,
        "opacity": 0.8,
        "timing": 475,
      },
    ],
    "blend": "difference",
    "blur": 2,
    "color": undefined,
    "dia": 200,
    "easing": "ease",
    "height": undefined,
    "opacity": 0.2,
    "timing": 20,
    "width": undefined,
    "zindex": 1,
  },
}
`;
